src/config.c

```diff
@@ -51,6 +51,12 @@ static int handler(void *user, const char *section, const char *name,
 		} else if (MATCH("postgres", "target_database")) {
 			ini_config->postgres_config->target_database =
 				strdup(value);
+		} else if (MATCH("postgres", "backup_type")) {
+			if (strcmp("schema", value) == 0)
+				ini_config->postgres_config->backup_type =
+					SCHEMA;
+			if (strcmp("full", value) == 0)
+				ini_config->postgres_config->backup_type = FULL;
```

Comment:
If I add the condition and `return 0` here (since the handler needs to `return 0` on failure), then it messes up
the error handling for parsing. It produces an `parse line` error with the wrong number of line (since i returned outside of the normal flow of the handler).

Reply:
There is no reason to return, it will return on it's own after the parsing is done.. I only mentioned printing to stderr.

Reply:
If I just print to stderr,  the flow continues and we actually replicate `full` though.

Reply:
The condition I provided prints to STDERR if the value is **not `empty`**, **not `schema`**, and **not `full`**.

Reply:
I.e.,

```ini
[postgres]
backup_type = unknown
```

Reply:
Yes but we continue the process of replicating if we dont return there, even with a wrongly provided `backup_type`.

Reply:
There are two [handler implementations exported](https://github.com/benhoyt/inih/blob/5cc5e2c24642513aaa5b19126aad42d0e4e0923e/ini.h#L51-L58) from `libinih`. If we use the implementation with the `int lineno` argument we should be OK.

Reply:
You mean without returning, or with `return lineno` ? Because neither work.